userspace/libsinsp/parsers.cpp

```diff
@@ -4499,6 +4499,46 @@ void sinsp_parser::parse_setgid_exit(sinsp_evt *evt)
 	}
 }
 
+static std::string convert_to_string(Json::ValueType other) {
+	std::string value_type("Not a valid json type");
+	switch(other) {
+	case Json::ValueType::nullValue:
+		return "nullValue";
+	case Json::ValueType::intValue:
+		return "intValue";
+	case Json::ValueType::uintValue:
+		return "uintValue";
+	case Json::ValueType::realValue:
+		return "realValue";
+	case Json::ValueType::stringValue:
+		return "stringValue";
+	case Json::ValueType::booleanValue:
+		return "booleanValue";
+	case Json::ValueType::arrayValue:
+		return "arrayValue";
+	case Json::ValueType::objectValue:
+		return "objectValue";
+	default:
+		break;
+	}
+	return value_type;
+}
+
+bool sinsp_parser::check_is_convertible(const Json::Value& value, Json::ValueType other, std::string field, bool log_message)
+{
+	if(!value.isConvertibleTo(other)) {
+		std::string value_as_string = value.isConvertibleTo(Json::stringValue) ? value.asString() : "value not convertible to string";
+		std::string err_msg = "Unable to convert json value '" + value_as_string + "' of type '" + convert_to_string(value.type()) + "' into type '" + convert_to_string(other) + "' for the field: '" + field +"'";
```

Comment:
This is totally not a hot path but still we generate the whole message even if we just drop it during logging. Can we move the string formatting to the macro invocations below? I think we'll live with the `convert_to_string()` calls :)

Reply:
done